assets/js/modules/adsense/components/dashboard/AdBlockingRecoveryWidget.test.js

```diff
@@ -0,0 +1,303 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import AdBlockingRecoveryWidget from './AdBlockingRecoveryWidget';
+import {
+	act,
+	fireEvent,
+	render,
+	createTestRegistry,
+	provideSiteInfo,
+	provideUserAuthentication,
+	provideModules,
+	unsubscribeFromAll,
+} from '../../../../../../tests/js/test-utils';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED,
+	MODULES_ADSENSE,
+} from '../../datastore/constants';
+import {
+	ACCOUNT_STATUS_PENDING,
+	ACCOUNT_STATUS_READY,
+	SITE_STATUS_ADDED,
+	SITE_STATUS_READY,
+} from '../../util';
+
+describe( 'AdSenseConnectCTA', () => {
+	let registry;
+	const validSettings = {
+		accountID: 'pub-12345678',
+		clientID: 'ca-pub-12345678',
+		useSnippet: false,
+		accountStatus: ACCOUNT_STATUS_READY,
+		siteStatus: SITE_STATUS_READY,
+		adBlockingRecoverySetupStatus:
+			AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED,
+	};
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		provideSiteInfo( registry );
+		provideUserAuthentication( registry );
+		provideModules( registry, [
+			{
+				active: true,
+				connected: true,
+				slug: 'adsense',
+			},
+		] );
+		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'before click', () => {
+		it.each( [
+			[
+				'the Adsense module is not connected',
+				ACCOUNT_STATUS_PENDING,
+				SITE_STATUS_READY,
+				'',
+				false,
+				false,
+				1684145866,
+			],
+			[
+				'notification is dismissed',
+				ACCOUNT_STATUS_PENDING,
+				SITE_STATUS_READY,
+				'',
+				true,
+				true,
+				1684145866,
+			],
+			[
+				'the Adsense account status is not ready',
+				ACCOUNT_STATUS_PENDING,
+				SITE_STATUS_READY,
+				'',
+				true,
+				false,
+				1684145866,
+			],
+			[
+				'the Adsense site status is not ready',
+				ACCOUNT_STATUS_READY,
+				SITE_STATUS_ADDED,
+				'',
+				true,
+				false,
+				1684145866,
+			],
+			[
+				'the Ad blocking recovery status is an empty string',
+				ACCOUNT_STATUS_READY,
+				SITE_STATUS_READY,
+				'',
+				true,
+				false,
+				1684145866,
+			],
+			[
+				'the setup completed timestamp is less than three weeks',
+				ACCOUNT_STATUS_READY,
+				SITE_STATUS_READY,
+				AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED,
+				true,
+				false,
+				1686836242,
```

Comment:
Wouldn't this value be _more_ than three weeks ago anyway in a few days at which point the test would start to fail?

Reply:
Good catch 👍

Reply:
I have set it to 2100. Let's see if Site Kit lives till then or we 😉